### PR TITLE
ADCM-2692 Refactor UI tests: test_configs

### PR DESCRIPTION
**Pull request by annademina**

### CODE REVIEW

**Comment by annademina** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @annademina! Build has been launched!

File: tests/ui_tests/app/page/common/configuration/page.py [hunk 1]
```diff
@@ -168,37 +168,61 @@ def assert_input_value_is(
...
+        :param expected_value: Value expected to be in input field
+        :param display_name: Config field display name
+        """
+
+        def _assert_value():
+            input_value = [
+                v.get_attribute("value")
+                for v in self.find_children(self.get_config_row(display_name), self.locators.ConfigRow.input)
+            ]
             assert expected_value == input_value, f'Expected value was {expected_value} but presented is {input_value}'
```
Comment (Sealwing) [inline tests/ui_tests/app/page/common/configuration/page.py:226]: Maybe add type annotation to `expected_value` and highlight that it's a list?

File: tests/ui_tests/app/page/common/configuration/page.py [hunk 2]
```diff
@@ -348,6 +372,14 @@ def check_invalid_value_message(self, error_message: str):
     def is_save_btn_disabled(self):
         return self.find_element(self.locators.save_btn).get_attribute("disabled") == 'true'
 
+    @allure.step("Check save button status")
+    def check_save_btn_state(self, expected_state: bool):
+        assert (
+            not (self.is_save_btn_disabled()) == expected_state
+        ), f'Save button should{" not " if expected_state is False else " "}be disabled'
+        if expected_state:
+            self.save_config()
```
Comment (Sealwing) [inline tests/ui_tests/app/page/common/configuration/page.py:381]: This method isn't only checking status of save button

**Review by Sealwing — CHANGES_REQUESTED** [review]

**Review by Sealwing — APPROVED** [review]

File: tests/ui_tests/app/helpers/configs_generator.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    if config['required'] and not config['default']:
+        expected_result['save'] = False
+        expected_result['alerts'] = True
+    else:
+        expected_result['save'] = True
+        expected_result['alerts'] = False
+    if config['read_only']:
+        expected_result['save'] = False
+    if config['ui_options']['invisible']:
+        expected_result['save'] = False
```
[the remark this answers is not in the document]
Comment (annademina) [reply tests/ui_tests/app/helpers/configs_generator.py:72]: changed

File: tests/ui_tests/app/helpers/configs_generator.py [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+    expected_result = {
+        'group_visible': not group_config['ui_options']['invisible'],
+        'editable': not group_config['read_only'],
+        "content": group_config['default'],
+    }
+    if group_config['required'] and not group_config['default']:
+        expected_result['alerts'] = True
+    else:
+        expected_result['alerts'] = False
```
[the remark this answers is not in the document]
Comment (annademina) [reply tests/ui_tests/app/helpers/configs_generator.py:161]: changed, thanks)

File: tests/ui_tests/app/page/common/configuration/page.py (hunk 1 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/configuration/page.py]: added

File: tests/ui_tests/app/page/common/configuration/page.py (hunk 2 above)
Comment (annademina) [reply tests/ui_tests/app/page/common/configuration/page.py:385]: renamed

**Comment by annademina** [comment]
bender build

**Comment by annademina** [comment]
bender build

**Comment by annademina** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @annademina! Build has been launched!

**Review by alexterent — APPROVED** [review]

**Pull request merged**